lib/Converter/FunctionConverter.cpp

```diff
@@ -159,6 +163,39 @@ void FunctionConverter::convert() {
   function_.verify();
 }
 
+bool FunctionConverter::canConvert(const Tensor &tensor, TypeRef dstTy) const {
+  // The default implementation of convertTensor uses Tensor::converToType
+  // which does not support quantization for now.
+  return !dstTy->isQuantizedType() && !tensor.getType().isQuantizedType();
+}
+
+void FunctionConverter::convertTensor(Tensor &tensor, TypeRef dstTy) {
+  assert(canConvert(tensor, dstTy) && "We shouldn't call this method");
+  assert(tensor.dims() == dstTy->dims() &&
+         "Conversion should only change the values, not the shape");
+  tensor.convertToType(dstTy->getElementType());
+}
+
+NodeValue FunctionConverter::convertConstant(Constant &constant,
+                                             TypeRef dstTy) {
+  const Tensor &tensor = constant.getPayload();
+  if (!canConvert(tensor, dstTy)) {
+    return NodeValue();
+  }
+
+  Constant *constantToBeModified = &constant;
+  if (!constant.hasOneUse()) {
+    // Could be a simple clone, but storage classes don't support cloning for
+    // now.
+    constantToBeModified =
+        mod_.createConstant(constant.getType(), constant.getName());
+  }
+  Tensor &tensorToBeModified = constantToBeModified->getPayload();
```

Comment:
Don't you need to assign the the new Constant's payload to the old one, or am I missing something? Also doesn't look like you're checking any values for this in the unit test added, so perhaps that should be added?

Reply:
My phone doesn’t show me the context but IIRC the following line convertTensor modifies the tensor given by reference.
Maybe I should make it more obvious.
Idea?

Reply:
`convertTensor()` modifies the tensor by reference, but the tensor's original contents (i.e. the contents from `constant`) don't appear to ever be assigned to `tensorToBeModified`. It looks to me like there are no references to the original `constant` after `constantToBeModified` is created, i.e. its backing payload tensor is probably left zero-initialized.

Reply:
Got it!

Reply:
I used clone previously before realizing it didn't work and stupidly just replaced it with createConstant which is not enough obviously.

Reply:
Done